src/Datasets/Dataset.php

```diff
@@ -290,6 +291,30 @@ public function apply(Transformer $transformer) : self
         return $this;
     }
 
+    /**
+     * Reverse a transformation to the dataset.
+     *
+     * @param \Rubix\ML\Transformers\Transformer $transformer
+     * @throws \Rubix\ML\Exceptions\RuntimeException
+     * @return static
+     */
+    public function reverse(Transformer $transformer) : self
+    {
+        if (!$transformer instanceof Bidirectional) {
+            throw new RuntimeException('Transformer does not support reverse transformation.');
+        }
+
+        if ($transformer instanceof Stateful) {
```

Comment:
If the transformer is Stateful and is not fitted, should it throw an exception instead? In order to reverse the transformation we'd need the information calculated at fitting no?

Reply:
Makes sense. Also covers some of the scenarios when `reverseTransform` is called before `transform`.